lib/landing_session.js

```diff
@@ -167,16 +186,42 @@ class LandingSession extends Session {
 
   async apply() {
     const { cli } = this;
+
+    // Bail if another landing session is currently in progress.
     if (!this.isApplying()) {
-      cli.warn('This session can not proceed to apply patches, ' +
-        'run `git node land --abort`');
+      cli.warn('Landing session already in progress - ' +
+        'to start a new one run `git node land --abort`');
       return;
     }
     await this.tryResetBranch();
 
     const patch = await this.downloadAndPatch();
+
+    const cleanLint = await this.validateLint(patch);
+    if (!cleanLint) {
+      const tryFixLint = await cli.prompt(
+        'Try fixing JS lint with \'make lint-js-fix\'?');
+      if (tryFixLint) {
+        await runAsync('make', ['lint-js-fix']);
+      }
+
+      const correctedLint = await cli.prompt('Corrected all lint errors?');
+      if (correctedLint) {
```

Comment:
:thinking: shouldn't we run `this.validateLint(patch);` again to make sure before committing or it will be checked later down the line?

Reply:
@lundibundi we can't really - it validates the pre-downloaded patch from upstream which can't be updated per se.